dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -631,105 +568,64 @@ abstract class DLCWallet
-      dlcAcceptDbs <- dlcAcceptDAO.findByDLCId(dlcId)
-      dlcAcceptFOpt = {
-        dlcAcceptDbs.headOption.map { case dlcAcceptDb =>
-          logger.debug(
-            s"DLC Accept (${dlcId.hex}) has already been made, returning accept")
-          for {
-            fundingInputs <-
-              dlcInputsDAO.findByDLCId(dlcId, isInitiator = false)
-            prevTxs <-
-              transactionDAO.findByTxIdBEs(fundingInputs.map(_.outPoint.txIdBE))
-            outcomeSigsDbs <- dlcSigsDAO.findByDLCId(dlcId)
-            refundSigsDb <- dlcRefundSigDAO.read(dlcId)
-          } yield {
-            val inputRefs =
-              DLCTxUtil.matchPrevTxsWithInputs(fundingInputs, prevTxs)
-
-            dlcAcceptDb.toDLCAccept(offer.tempContractId,
-                                    inputRefs,
-                                    outcomeSigsDbs.map { db =>
-                                      db.sigPoint -> db.accepterSig
-                                    },
-                                    refundSigsDb.get.accepterSig)
-          }
-        }
-      }
-    } yield {
-      dlcAcceptFOpt
-    }
-
-    resultNestedF.flatMap {
-      case Some(f) => f.map(Some(_))
-      case None    => Future.successful(None)
-    }
-  }
-
   private def createNewDLCAccept(
       collateral: CurrencyUnit,
       offer: DLCOffer): Future[DLCAccept] = {
     logger.info(
       s"Creating DLC Accept for tempContractId ${offer.tempContractId.hex}")
 
-    val dlcDbAccountDbF: Future[(DLCDb, DLCOfferDb, AccountDb)] =
-      initDLCForAccept(offer)
-    for {
-      (dlc, _, account) <- dlcDbAccountDbF
-      (txBuilder, spendingInfos) <- fundRawTransactionInternal(
-        destinations = Vector(TransactionOutput(collateral, EmptyScriptPubKey)),
-        feeRate = offer.feeRate,
-        fromAccount = account,
-        fromTagOpt = None,
-        markAsReserved = true
-      )
-      serialIds = DLCMessage.genSerialIds(
-        spendingInfos.size,
-        offer.fundingInputs.map(_.inputSerialId))
-      utxos = spendingInfos.zip(serialIds).map { case (utxo, id) =>
-        DLCFundingInput.fromInputSigningInfo(
-          utxo,
-          id,
-          TransactionConstants.enableRBFSequence)
-      }
-
-      changeSPK = txBuilder.finalizer.changeSPK
-      changeAddr = BitcoinAddress.fromScriptPubKey(changeSPK, networkParameters)
-
-      bip32Path = BIP32Path(
-        account.hdAccount.path ++ Vector(BIP32Node(0, hardened = false),
-                                         BIP32Node(dlc.keyIndex,
-                                                   hardened = false)))
+    val accountF = getDefaultAccountForType(AddressType.SegWit)
 
-      privKeyPath = HDPath.fromString(bip32Path.toString)
-      fundingPrivKey =
-        keyManager.toSign(privKeyPath)
-
-      dlcPubKeys = calcDLCPubKeys(account.xpub, dlc.changeIndex, dlc.keyIndex)
+    val txBuilderAndSpendingInfosF: Future[(
```

Comment:
I don't think we really want to execute these `Future`s in parallel.

Reply:
What would you like it synchronized with?